libcudacxx/include/cuda/std/__type_traits/type_list.h

```diff
@@ -394,6 +456,52 @@ using __type_fold_right = __type_call<_List, detail::__type_fold_right_fn<_Init,
 template <class _List, class _Init, class _Fn>
 using __type_fold_left = __type_call<_List, detail::__type_fold_left_fn<_Init, _Fn>>;
 
+namespace detail
```

Comment:
Everything here is already ugly, is there a necessity to using a detail namespace? If so please use `__detail`, because macros are evil